bee-network/bee-gossip/src/network/host.rs

```diff
@@ -209,13 +206,36 @@ async fn dial_peer(swarm: &mut Swarm<SwarmBehaviour>, peer_id: PeerId, peerlist:
     // We just checked, that the peer is fine to be dialed.
     let PeerInfo {
         address: addr, alias, ..
-    } = peerlist.0.read().await.info(&peer_id).unwrap();
-
-    debug!("Dialing peer: {} ({}).", alias, alias!(peer_id));
+    } = peerlist.0.read().await.info(&peer_id).expect("peer must exist");
+
+    let mut dial_attempt = 0;
+
+    peerlist
+        .0
+        .write()
+        .await
+        .update_metrics(&peer_id, |m| {
+            m.num_dials += 1;
+            dial_attempt = m.num_dials;
+        })
+        .expect("peer must exist");
+
+    debug!(
+        "Dialing peer: {} ({}) attempt: #{}.",
+        alias,
+        alias!(peer_id),
+        dial_attempt
+    );
 
     // TODO: We also use `Swarm::dial_addr` here (instead of `Swarm::dial`) for now. See if it's better to change
     // that.
     Swarm::dial_addr(swarm, addr).map_err(|e| Error::DialingPeerFailed(peer_id, e))?;
 
     Ok(())
 }
+
+async fn hang_up(swarm: &mut Swarm<SwarmBehaviour>, peer_id: PeerId) {
```

Comment:
Doesn't seem to need the async ?
Also, you don't do a function for the other events, I wouldn't for consistency.

Reply:
No, it's not needed atm (in fact it won't ever be due to #1174), but decided to add it because the "nature" of those functions is the same. It's not unlikely that in the (in fact not happening) future `hang_up` may require access to the `peerlist` which is protected by async lock.

The second sentence I don't understand. This is a command handler just as `dial_addr` and `dial_peer`. What's different here is that I don't return an error (which I could do for consistency, but decided against it to keep the changes as small as possible in this PR)

Reply:
Yeah I'm just not sure if the compiler will create a state machine for it or not. In doubt, if not needed I wouldn't put it. Ignore the second sentence, I confused myself 🙃

Reply:
That's actually something worth to know. I will try to find something on it online. I had my hopes up that the compiler would be smart enough to remove syntactic sugar automatically, but here I may be wrong.